ui/src/treeDataProvider/AzureTreeDataProvider.ts

```diff
@@ -220,6 +221,25 @@ export class AzureTreeDataProvider<TRoot = ISubscriptionRoot> implements IAzureT
         return element.parent;
     }
 
+    public async getSubscriptionPromptStep(wizardContext: Partial<types.ISubscriptionWizardContext>): Promise<types.AzureWizardPromptStep<types.ISubscriptionWizardContext> | undefined> {
+        const subscriptions: AzureTreeItem<ISubscriptionRoot>[] = await this.ensureRootTreeItems();
+        if (subscriptions.length === 1) {
+            assignRootToWizardContext(wizardContext, subscriptions[0].root);
+            return undefined;
+        } else {
+            // tslint:disable-next-line: no-var-self
+            const tree: AzureTreeDataProvider<TRoot> = this;
+            class SubscriptionPromptStep extends AzureWizardPromptStep<types.ISubscriptionWizardContext> {
+                public async prompt(): Promise<void> {
+                    const ti: AzureTreeItem<TRoot | ISubscriptionRoot> = await tree.promptForRootTreeItem(SubscriptionTreeItem.contextValue);
+                    assignRootToWizardContext(wizardContext, <types.ISubscriptionRoot>ti.root);
```

Comment:
What if the user cancels the prompt and `ti` is undefined?  Does `promptForRootTreeItem` use our `UserCancelledError` handling?

Reply:
Yeah it'll throw a `promptForRootTreeItem`, not return undefined.